arbnode/batch_poster.go

```diff
@@ -262,12 +263,13 @@ func NewBatchPoster(dataPosterDB ethdb.Database, l1Reader *headerreader.HeaderRe
 // contain reverted batch_poster transaction.
 // It returns true if it finds batch posting needs to halt, which is true if a batch reverts
 // unless the data poster is configured with noop storage which can tolerate reverts.
-func (b *BatchPoster) checkReverts(ctx context.Context, from, to int64) (bool, error) {
-	if from > to {
-		return false, fmt.Errorf("wrong range, from: %d is more to: %d", from, to)
+// From must be a pointer to the starting block, which is updated after each block is checked for reverts
```

Comment:
I think we should just make "nextToCheck" batchposter field now that we want to update it. That will be more readable than passing pointer and updating it in place.